internal/pkg/override/cdk.go

```diff
@@ -172,6 +181,95 @@ func (cdk *CDK) cleanUp(in []byte) ([]byte, error) {
 	return out.Bytes(), nil
 }
 
+type packageManager struct {
+	name     string
+	lockFile string
+}
+
+var packageManagers = []packageManager{ // Alphabetically sorted based on name.
+	{
+		name:     "npm",
+		lockFile: "package-lock.json",
+	},
+	{
+		name:     "yarn",
+		lockFile: "yarn.lock",
+	},
+}
+
+func (cdk *CDK) installedPackageManager() ([]string, error) {
+	var lookUpExecutableErrs []error
+	var installed []string
+	for _, candidate := range packageManagers {
+		if _, err := cdk.exec.LookPath(candidate.name); err == nil {
+			installed = append(installed, candidate.name)
+		} else {
+			lookUpExecutableErrs = append(lookUpExecutableErrs, &errorExecutableNotFound{
+				executable: candidate.name,
+				error:      err,
+			})
+		}
+	}
```

Comment:
seems like the only error case we care about is `len(installed) == 0` - should we remove this case and just return an error if we don't find _any_?

Reply:
Good point, I addressed this by doing the following instead:
```go
func (cdk *CDK) installedPackageManagers() ([]string, error) {
	var installed []string
	for _, candidate := range packageManagers {
		if _, err := cdk.exec.LookPath(candidate.name); err == nil {
			installed = append(installed, candidate.name)
		} else if !errors.Is(err, exec.ErrNotFound) { // Return unknown error! Swallow ErrNotFound.
			return nil, err
		}
	}
	return installed, nil
}


func (cdk *CDK) packageManager() (string, error) {
	installed, err := cdk.installedPackageManagers()
	if err != nil { // Handle unknown error.
		return "", err
	}
	if len(installed) == 0 {
		return "", &errPackageManagerUnavailable{}
	}
```